domain/secret/modelmigration/export.go

```diff
@@ -0,0 +1,257 @@
+				ExpireTime: rev.ExpireTime,
+			}
+
+			content, ok := allSecrets.Content[md.URI.ID][rev.Revision]
+			if ok && len(content) > 0 {
+				revArg.Content = make(secrets.SecretData)
+				for k, v := range content {
+					revArg.Content[k] = fmt.Sprintf("%v", v)
+				}
+			}
+			if rev.ValueRef != nil {
+				revArg.ValueRef = &description.SecretValueRefArgs{
+					BackendID:  rev.ValueRef.BackendID,
+					RevisionID: rev.ValueRef.RevisionID,
+				}
+			}
+			revisionArgsByID[md.URI.ID] = append(revisionArgsByID[md.URI.ID], revArg)
+		}
+
+		arg := description.SecretArgs{
+			ID:                     md.URI.ID,
+			Version:                md.Version,
+			Description:            md.Description,
+			Label:                  md.Label,
+			RotatePolicy:           md.RotatePolicy.String(),
+			AutoPrune:              md.AutoPrune,
+			Owner:                  ownerTag,
+			Created:                md.CreateTime,
+			Updated:                md.UpdateTime,
+			NextRotateTime:         md.NextRotateTime,
+			LatestRevisionChecksum: md.LatestRevisionChecksum,
+			Revisions:              revisionArgsByID[md.URI.ID],
+			ACL:                    accessArgsByID[md.URI.ID],
+			Consumers:              consumersByID[md.URI.ID],
+			RemoteConsumers:        remoteConsumersByID[md.URI.ID],
+		}
+		model.AddSecret(arg)
+	}
+
+	for _, rs := range allSecrets.RemoteSecrets {
+		consumerTag, err := accessorTagFromAccessor(rs.Accessor)
+		if err != nil {
+			// Should never happen.
+			return errors.Trace(err)
+		}
+		arg := description.RemoteSecretArgs{
+			ID:              rs.URI.ID,
+			SourceUUID:      rs.URI.SourceUUID,
+			Consumer:        consumerTag,
+			Label:           rs.Label,
+			CurrentRevision: rs.CurrentRevision,
+			LatestRevision:  rs.LatestRevision,
+		}
+		model.AddRemoteSecret(arg)
+	}
+	return nil
+}
+
+func (e *exportOperation) collateConsumers(consumers map[string][]service.ConsumerInfo) (map[string][]description.SecretConsumerArgs, error) {
+	consumersByID := make(map[string][]description.SecretConsumerArgs)
+	for id, consumers := range consumers {
+		for _, info := range consumers {
+			consumerTag, err := accessorTagFromAccessor(info.Accessor)
+			if err != nil {
+				// Should never happen.
+				return nil, errors.Trace(err)
+			}
+			consumerArg := description.SecretConsumerArgs{
+				Consumer:        consumerTag,
+				Label:           info.Label,
+				CurrentRevision: info.CurrentRevision,
+			}
+			consumersByID[id] = append(consumersByID[id], consumerArg)
+		}
+	}
+	return consumersByID, nil
+}
+
+func (e *exportOperation) collateRemoteConsumers(consumers map[string][]service.ConsumerInfo) (map[string][]description.SecretRemoteConsumerArgs, error) {
+	consumersByID := make(map[string][]description.SecretRemoteConsumerArgs)
+	for id, consumers := range consumers {
+		for _, info := range consumers {
+			consumerTag, err := accessorTagFromAccessor(info.Accessor)
+			if err != nil {
+				// Should never happen.
+				return nil, errors.Trace(err)
```

Comment:
Same here.

Reply:
Because this error will never happen in production (will be caught in unit tests), IMO there's little value in annotating it. But will do it.